nan.h

```diff
@@ -1527,26 +1755,28 @@ class NanObjectWrap : public node::ObjectWrap {
 template <typename T, typename P>
 class NanWeakCallbackData {
  public:  // constructors
+  inline ~NanWeakCallbackData();
+
+ private:  // constructors
+  NAN_DISALLOW_ASSIGN_COPY_MOVE(NanWeakCallbackData)
+
+ public:
   typedef void (*Callback)(
       NanWeakCallbackData & data  // NOLINT(runtime/references)
       );
-  NanWeakCallbackData(v8::Handle<T> handle, P* param, Callback cb)
-    : parameter(param), callback(cb) {
-    NanAssignPersistent(persistent, handle);
-    Revive();
+  NanWeakCallbackData(v8::Persistent<T> *handle, P* param, Callback cb)
+    : persistent(), parameter(param), callback(cb) {
+    std::memcpy(&persistent, handle, sizeof(persistent));
```

Comment:
This is to copy a (new-style) `v8::Persistent` without duplicating the storage cell, aka old-style copying. Need the actual `Persistent` handle to be able to access it from the weak callback, as I cannot reconstruct a Persistent handle in any way. Can't use a reference or pointer to the original hanlde, as it will go out of scope and be deallocated (not touching the underlying global storage cell) until the weak callback is called.

Is this safe? Should be. Is there any better way?